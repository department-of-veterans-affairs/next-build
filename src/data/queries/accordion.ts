import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAccordion } from '@/types/drupal/paragraph'
import { Accordion } from '@/types/formatted/accordion'

export const formatter: QueryFormatter<ParagraphAccordion, Accordion> = (
  entity: ParagraphAccordion
) => {
  return {
    id: entity.id,
    header: entity.field_header || null,
    html: entity.field_rich_wysiwyg.processed,
  }
}
