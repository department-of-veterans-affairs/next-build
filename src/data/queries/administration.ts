import { FieldAdministration } from '@/types/drupal/field_type'

export interface Administration {
  entityId: number
  name: string | null
}

/**
 * Formats a Drupal administration field into our standardized Administration interface.
 * 
 * In our Drupal content, the `field_administration` field points to a taxonomy Term
 * that is currently called "Section", but "Section" is only a human label added to
 * it in the Drupal interface, and that name has evolved over time. It was originally,
 * "administration" (and thus the name of the field). Since the only machine name of
 * it is `field_administration`, we'll continue to call it "administration" in our code.
 */
export const formatAdministration = (
  administration: FieldAdministration | null | undefined
): Administration => ({
  entityId: administration?.drupal_internal__tid || null, // "t" is for taxonomy term
  name: administration?.name || null,
})
