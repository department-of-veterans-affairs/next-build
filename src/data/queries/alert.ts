// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter } from 'next-drupal-query'
import { Alert } from '@/types/formatted/alert'

export const formatter: QueryFormatter<BlockAlert, Alert> = (
  entity: BlockAlert
) => {
  return {
    alertType:
      entity.field_alert_type === 'information'
        ? 'info'
        : entity.field_alert_type,
    id: entity.id,
    title: entity.field_alert_title,
    content: {
      header: entity.field_alert_content.field_text_expander,
      text: entity.field_alert_content.field_wysiwyg.processed,
    },
  }
}
