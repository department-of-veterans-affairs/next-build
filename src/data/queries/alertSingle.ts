import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAlertSingle } from '@/types/drupal/paragraph'
import {
  AlertSingle,
  AlertBlock,
  AlertNonReusable,
} from '@/types/formatted/alert'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { queries } from '.'

// TODO:
//
// EDIT:
// It seems *maybe* that next-drupal-query is designed to handle this already.
// You can pass an `id` to queries.getParams().
// E.g.
// queries.getParams('paragraph--alert_single'))
//
// We don't need this here:
// export const params: QueryParams<null> = () => {
//   return queries
//     .getParams()
//     .addInclude([
//       'field_alert_block_reference',
//       'field_alert_block_reference.field_alert_content',
//       'field_alert_non_reusable_ref',
//       'field_alert_non_reusable_ref.field_va_paragraphs',
//     ])
// }
//
// ...but it seems like we might want to define
// paragraph-specific `includes` within the paragraph's
// query file (this file), and then somehow reference these in
// the node query when we need to include fields of a specific
// paragraph type.
//
// E.g.
// export const include = [
//   'field_alert_block_reference',
//   'field_alert_block_reference.field_alert_content',
//   'field_alert_non_reusable_ref',
//   'field_alert_non_reusable_ref.field_va_paragraphs',
// ]
//
// /* Some util file*/
// export const getIncludedSubFields = (
//   parentField: string,
//   subFields: string[]
// ) => [parentField, ...subFields.map((subField) => `${parentField}.${subField}`)]
//
// /* src/data/queries/resourcesSupport.ts */
// import { include } from '@/data/queries/alertSingle'
// export const params: QueryParams<null> = () => {
//   return queries.getParams().addInclude([
//     ...getIncludedSubFields('field_alert_single', include)
//   ])
// }

export const formatter: QueryFormatter<ParagraphAlertSingle, AlertSingle> = (
  entity: ParagraphAlertSingle
) => {
  return {
    type: entity.type as AlertSingle['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertSelection:
      entity.field_alert_selection as AlertSingle['alertSelection'],
    blockReference: queries.formatData(
      'block--alert',
      entity.field_alert_block_reference
    ) as AlertBlock,
    nonReusableRef: formatParagraph(
      entity.field_alert_non_reusable_ref
    ) as AlertNonReusable,
  }
}
