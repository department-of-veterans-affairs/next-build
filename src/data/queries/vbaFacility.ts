import { QueryData, QueryFormatter, QueryParams } from 'next-drupal-query'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { NodeVbaFacility, NodeVbaService } from '@/types/drupal/node'
import { VbaFacility } from '@/types/formatted/vbaFacility'
import { Wysiwyg } from '@/types/formatted/wysiwyg'
import { getHtmlFromField } from '@/lib/utils/getHtmlFromField'
import { getHtmlFromDrupalContent } from '@/lib/utils/getHtmlFromDrupalContent'
import { getNestedIncludes } from '@/lib/utils/queries'
import { formatter as formatImage } from '@/data/queries/mediaImage'
import { formatter as formatFeaturedContent } from '@/data/queries/featuredContent'
import { formatter as formatAccordionItem } from '@/data/queries/accordion'
import { formatter as formatPhone } from '@/data/queries/phoneNumber'
import { formatter as formatEmail } from '@/data/queries/emailContact'
import { createPhoneLinks } from '@/lib/utils/createPhoneLinks'
import { PAGE_SIZES } from '@/lib/constants/pageSizes'
import {
  PARAGRAPH_RESOURCE_TYPES,
  RESOURCE_TYPES,
} from '@/lib/constants/resourceTypes'
import { ExpandedStaticPropsContext } from '@/lib/drupal/staticProps'
import {
  entityBaseFields,
  fetchSingleEntityOrPreview,
  fetchAndConcatAllResourceCollectionPages,
} from '@/lib/drupal/query'
import { PhoneContact } from '@/types/formatted/contactInfo'

const isPublished = (entity: { status: boolean }) => entity.status === true

// Define the query params for fetching node--vba_facility.
export const params: QueryParams<null> = () => {
  return new DrupalJsonApiParams().addInclude([
    'field_media',
    'field_media.image',
    'field_prepare_for_visit',
    ...getNestedIncludes(
      'field_local_spotlight',
      PARAGRAPH_RESOURCE_TYPES.FEATURED_CONTENT
    ),
  ])
}
export const serviceParams: QueryParams<string> = (facilityId: string) => {
  return new DrupalJsonApiParams()
    .addInclude([
      'field_office',
      'field_service_name_and_descripti',
      'field_service_location',
      'field_service_location.field_service_location_address',
      'field_service_location.field_other_phone_numbers',
      'field_service_location.field_phone',
      'field_service_location.field_email_contacts',
    ])
    .addFilter('field_office.id', facilityId)
}
// Define the option types for the data loader.
export type VbaFacilityDataOpts = {
  id: string
  context?: ExpandedStaticPropsContext
}
type VbaFacilityData = {
  entity: NodeVbaFacility
  services: NodeVbaService[]
}
// Implement the data loader.
export const data: QueryData<VbaFacilityDataOpts, VbaFacilityData> = async (
  opts
): Promise<VbaFacilityData> => {
  const entity = (await fetchSingleEntityOrPreview(
    opts,
    RESOURCE_TYPES.VBA_FACILITY,
    params
  )) as NodeVbaFacility
  const { data: services } =
    await fetchAndConcatAllResourceCollectionPages<NodeVbaService>(
      RESOURCE_TYPES.VBA_SERVICE,
      serviceParams(entity.id),
      PAGE_SIZES.MAX
    )

  return {
    entity,
    services,
  }
}
const getLinkType = (uri: string) => {
  const types = [
    { match: 'facebook.com', type: 'facebook' },
    { match: 'twitter.com', type: 'x' },
    { match: 'x.com', type: 'x' },
    { match: 'instagram.com', type: 'instagram' },
    { match: 'govdelivery.com', type: 'mail' },
    { match: 'flickr.com', type: 'flickr' },
    { match: 'youtube.com', type: 'youtube' },
  ]
  const found = types.find(({ match }) => uri.includes(match))
  return found ? found.type : null
}

export const formatter: QueryFormatter<VbaFacilityData, VbaFacility> = ({
  entity,
  services,
}) => {
  const featuredContent = [
    formatFeaturedContent({
      type: 'paragraph--featured_content',
      id: entity.field_cc_national_spotlight_1.target_id,
      field_section_header:
        entity.field_cc_national_spotlight_1.fetched.field_section_header[0]
          .value,
      field_description: {
        ...entity.field_cc_national_spotlight_1.fetched.field_description[0],
      },
      drupal_internal__id: null,
      drupal_internal__revision_id: null,
      langcode: null,
      status: true,
      field_cta: {
        type: 'paragraph--button',
        field_button_label:
          entity.field_cc_national_spotlight_1.fetched.field_cta[0]
            .field_button_label[0].value,
        field_button_link: {
          ...entity.field_cc_national_spotlight_1.fetched.field_cta[0]
            .field_button_link[0],
          options: [],
        },
        drupal_internal__id: parseInt(
          entity.field_cc_national_spotlight_1.fetched.field_cta[0].target_id
        ),
        drupal_internal__revision_id: parseInt(
          entity.field_cc_national_spotlight_1.fetched.field_cta[0]
            .target_revision_id
        ),
        langcode:
          entity.field_cc_national_spotlight_1.fetched.field_cta[0].langcode,
        status:
          entity.field_cc_national_spotlight_1.fetched.field_cta[0].status,
        id: null,
      },
    }),
    ...entity.field_local_spotlight.map((feature) => {
      return formatFeaturedContent(feature)
    }),
  ]
  return {
    ...entityBaseFields(entity),
    address: entity.field_address,
    ccBenefitsHotline: {
      type: PARAGRAPH_RESOURCE_TYPES.PHONE_CONTACT as PhoneContact['type'],
      label:
        entity.field_cc_benefits_hotline.fetched.field_phone_label[0]?.value ||
        null,
      number:
        entity.field_cc_benefits_hotline.fetched.field_phone_number[0]?.value ||
        null,
      extension:
        entity.field_cc_benefits_hotline.fetched.field_phone_extension[0]
          ?.value || null,
      id: entity.field_cc_benefits_hotline.target_id || null,
    },
    ccCantFindBenefits: entity.field_cc_cant_find_benefits
      ? {
          header:
            entity.field_cc_cant_find_benefits?.fetched?.field_section_header[0]
              ?.value || null,
          description: getHtmlFromField(
            entity.field_cc_cant_find_benefits?.fetched?.field_description[0]
          ),
          link: {
            label:
              entity.field_cc_cant_find_benefits?.fetched?.field_cta[0]
                ?.field_button_label[0]?.value || null,
            url:
              entity.field_cc_cant_find_benefits?.fetched?.field_cta[0]
                ?.field_button_link[0]?.uri || null,
          },
        }
      : null,
    ccGetUpdates: entity.field_cc_get_updates_from_vba
      ? {
          heading:
            entity.field_cc_get_updates_from_vba.fetched.field_section_header[0]
              .value || null,
          links: entity.field_cc_get_updates_from_vba.fetched.field_links.map(
            (link) => ({
              label: link.title,
              url: link.uri,
              type: getLinkType(link.uri),
            })
          ),
        }
      : null,
    ccVBAFacilityOverview: {
      type: PARAGRAPH_RESOURCE_TYPES.WYSIWYG as Wysiwyg['type'],
      html: getHtmlFromField(
        entity.field_cc_vba_facility_overview.fetched.field_wysiwyg[0]
      ),
      id: entity.field_cc_vba_facility_overview.target_id || null,
    },
    facilityLocatorApiId: entity.field_facility_locator_api_id,
    featuredContent: featuredContent,
    image: entity.field_media ? formatImage(entity.field_media) : null,
    officeHours: entity.field_office_hours,
    operatingStatusFacility: entity.field_operating_status_facility,
    operatingStatusMoreInfo: entity.field_operating_status_more_info
      ? getHtmlFromDrupalContent(entity.field_operating_status_more_info, {
          convertNewlines: true,
        })
      : null,
    prepareForVisit: entity.field_prepare_for_visit.map(formatAccordionItem),
    phoneNumber: entity.field_phone_number,
    allServices: services.map((service) => ({
      type: service.field_service_name_and_descripti?.field_vba_type_of_care,
      name: service.field_service_name_and_descripti?.name,
      facilityHeader:
        service.field_service_name_and_descripti?.field_facility_service_header,
      facilityDescription:
        service.field_service_name_and_descripti
          ?.field_facility_service_descripti,
      onlineSelfService: service.field_service_name_and_descripti
        ?.field_online_self_service
        ? {
            title:
              service.field_service_name_and_descripti.field_online_self_service
                .title,
            url: service.field_service_name_and_descripti
              .field_online_self_service.url,
          }
        : null,
      serviceDescription:
        service.field_service_name_and_descripti?.field_vba_service_descrip,
      serviceLocations: service.field_service_location.map((location) => ({
        fieldOfficeVisits: location.field_office_visits,
        fieldVirtualSupport: location.field_virtual_support,
        fieldApptIntroTextType: location.field_appt_intro_text_type,
        fieldApptIntroTextCustom: createPhoneLinks(
          location.field_appt_intro_text_custom
        ),
        appointmentPhoneNumbers: location.field_other_phone_numbers
          .filter(isPublished)
          .map(formatPhone),
        fieldOnlineSchedulingAvail: location.field_online_scheduling_avail,
        contactInfoPhoneNumbers: location.field_phone
          .filter(isPublished)
          .map(formatPhone),
        fieldEmailContacts: location.field_email_contacts
          .filter(isPublished)
          .map(formatEmail),
        fieldHours: location.field_hours,
        fieldOfficeHours: location.field_office_hours,
        fieldAdditionalHoursInfo: location.field_additional_hours_info,
        fieldUseMainFacilityPhone: location.field_use_main_facility_phone,
        fieldUseFacilityPhoneNumber: location.field_use_facility_phone_number,
        fieldServiceLocationAddress: location.field_service_location_address,
      })),
    })),
  }
}
