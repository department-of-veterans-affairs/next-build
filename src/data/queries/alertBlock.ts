// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter } from 'next-drupal-query'
import { AlertBlock, AlertType } from '@/types/formatted/alert'

export const formatter: QueryFormatter<BlockAlert, AlertBlock> = (
  entity: BlockAlert
) => {
  if (!entity) {
    return null
  }

  return {
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    id: entity.id,
    title: entity.field_alert_title,
    content: {
      header:
        entity.field_alert_content.field_text_expander ??
        entity.field_alert_heading,
      text: entity.field_alert_content?.field_wysiwyg?.processed ?? '',
    },
  }
}
