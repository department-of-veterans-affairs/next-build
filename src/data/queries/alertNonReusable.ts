// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter } from 'next-drupal-query'
import { AlertType, AlertNonReusable } from '@/types/formatted/alert'
import { ParagraphNonReusableAlert } from '@/types/drupal/paragraph'
import { formatParagraph } from '@/lib/drupal/paragraphs'

export const formatter: QueryFormatter<
  ParagraphNonReusableAlert,
  AlertNonReusable
> = (entity: ParagraphNonReusableAlert) => {
  if (!entity) {
    return null
  }

  return {
    type: entity.type as AlertNonReusable['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    heading: entity.field_alert_heading,
    paragraphs: entity.field_va_paragraphs.map(formatParagraph),
  }
}
