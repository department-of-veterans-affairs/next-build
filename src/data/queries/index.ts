import { createQueries } from 'next-drupal-query'
import * as NewsStory from './newsStory'
import * as NewsStoryTeaser from './newsStoryTeaser'
import * as StoryListing from './storyListing'
import * as QuestionAnswer from './questionAnswer'
import * as ExpandableText from './expandableText'
import * as LinkTeaser from './linkTeaser'
import * as MediaImage from './mediaImage'
import * as Banner from './banners'
import * as PersonProfile from './personProfile'
import * as Button from './button'
import * as AudienceTopics from './audienceTopics'
import * as Alert from './alert'
import * as EmailContact from './emailContact'
import * as BenefitsHub from './benefitsHub'
import * as Wysiwyg from './wysiwyg'
import * as StaticPathResources from './staticPathResources'
import * as HeaderFooter from './headerFooter'

export const queries = createQueries({
  // standard Drupal entity data queries
  'node--news_story': NewsStory,
  'node--news_story--teaser': NewsStoryTeaser,
  'node--story_listing': StoryListing,
  'node--q_a': QuestionAnswer,
  'node--person_profile': PersonProfile,
  'node--landing_page': BenefitsHub, // "Benefits Hub Landing Page"
  'paragraph--audience_topics': AudienceTopics,
  'paragraph--button': Button,
  'paragraph--email_contact': EmailContact,
  'paragraph--expandable_text': ExpandableText,
  'paragraph--link_teaser': LinkTeaser,
  'paragraph--rich_text_char_limit_1000': Wysiwyg,
  'paragraph--wysiwyg': Wysiwyg,
  'media--image': MediaImage,
  'block--alert': Alert,

  // Custom queries 
  'banner--alerts_lookup': Banner,
  'header-footer-data': HeaderFooter,

  // Static Path Generation
  'static-path-resources': StaticPathResources,
})
