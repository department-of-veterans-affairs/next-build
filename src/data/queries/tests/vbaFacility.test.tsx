/**
 * @jest-environment node
 */

import { queries } from '@/data/queries'
import mockData from '@/mocks/vbaFacility.mock.json'
import mockServicesData from '@/mocks/vbaFacilityService.mock.json'
import { params } from '../vbaFacility'

const VbaFacilityMock = { entity: mockData, services: [mockServicesData] }

// remove if this component does not have a data fetch
describe('DrupalJsonApiParams configuration', () => {
  test('params function sets the correct include fields', () => {
    const paramsInstance = params()
    const queryString = decodeURIComponent(paramsInstance.getQueryString())
    expect(queryString).toMatch(
      /include=field_media,field_media.image,field_prepare_for_visit,field_local_spotlight,field_local_spotlight.field_cta/
    )
  })
})

describe('VbaFacility formatData', () => {
  test('outputs formatted data', () => {
    expect(
      queries.formatData('node--vba_facility', VbaFacilityMock)
    ).toMatchSnapshot()
  })

  test('formats social link types', () => {
    const allSocialLinkMock = {
      ...VbaFacilityMock,
      entity: {
        ...VbaFacilityMock.entity,
        field_cc_get_updates_from_vba: {
          fetched: {
            field_section_header: [{ value: 'Get updates from VBA' }],
            field_links: [
              {
                title: 'Facebook',
                uri: 'https://www.facebook.com/veteransbenefits',
              },
              {
                title: 'Twitter',
                uri: 'https://twitter.com/VeteransBenefits',
              },
              {
                title: 'X',
                uri: 'https://x.com/VeteransBenefits',
              },
              {
                title: 'Instagram',
                uri: 'https://instagram.com/VeteransBenefits',
              },
              {
                title: 'Mailing List',
                uri: 'https://www.govdelivery.com/veteransbenefits',
              },
              {
                title: 'Flickr',
                uri: 'https://www.flickr.com/photos/veteransbenefits',
              },
              {
                title: 'YouTube',
                uri: 'https://www.youtube.com/veteransbenefits',
              },
            ],
          },
        },
      },
    }
    const formattedData = queries.formatData(
      'node--vba_facility',
      allSocialLinkMock
    )
    expect(formattedData.ccGetUpdates.links).toEqual([
      {
        label: 'Facebook',
        url: 'https://www.facebook.com/veteransbenefits',
        type: 'facebook',
      },
      {
        label: 'Twitter',
        url: 'https://twitter.com/VeteransBenefits',
        type: 'x',
      },
      { label: 'X', url: 'https://x.com/VeteransBenefits', type: 'x' },
      {
        label: 'Instagram',
        url: 'https://instagram.com/VeteransBenefits',
        type: 'instagram',
      },
      {
        label: 'Mailing List',
        url: 'https://www.govdelivery.com/veteransbenefits',
        type: 'mail',
      },
      {
        label: 'Flickr',
        url: 'https://www.flickr.com/photos/veteransbenefits',
        type: 'flickr',
      },
      {
        label: 'YouTube',
        url: 'https://www.youtube.com/veteransbenefits',
        type: 'youtube',
      },
    ])
  })
})
