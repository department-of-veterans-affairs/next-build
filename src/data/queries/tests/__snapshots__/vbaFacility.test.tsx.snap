// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VbaFacility formatData outputs formatted data 1`] = `
{
  "address": {
    "additional_name": null,
    "address_line1": "7305 N. Military Trail, Bldg 10",
    "administrative_area": "FL",
    "country_code": "US",
    "dependent_locality": null,
    "langcode": "en",
    "locality": "West Palm Beach",
    "postal_code": "33410",
  },
  "breadcrumbs": [
    {
      "options": [],
      "title": "Home",
      "uri": "https://va-gov-cms.ddev.site/",
    },
    {
      "options": [],
      "title": "St. Petersburg VA Regional Benefit Office",
      "uri": "https://va-gov-cms.ddev.site/st-petersburg-va-regional-benefit-office",
    },
    {
      "options": [],
      "title": "Veteran Readiness and Employment Office At West Palm Beach VA Medical Center",
      "uri": "https://va-gov-cms.ddev.site/st-petersburg-va-regional-benefit-office/locations/veteran-readiness-and-employment-office-at-west-palm-beach-va-medical-center",
    },
  ],
  "ccBenefitsHotline": {
    "extension": null,
    "id": "9be07ebb-f363-4780-af80-412b38a3b680",
    "label": "VA benefits hotline",
    "number": "800-827-1000",
    "type": "paragraph--phone_number",
  },
  "ccVBAFacilityOverview": {
    "html": "<p>We help Veterans, service members, and their families access VA benefits and services. Benefits we can help with include disability compensation, education benefits, life insurance, pensions, and home loans.</p>
",
    "id": "eecd106d-261c-4496-b449-8ce405945f2f",
    "type": "paragraph--wysiwyg",
  },
  "entityId": 4030,
  "entityPath": "/st-petersburg-va-regional-benefit-office/locations/veteran-readiness-and-employment-office-at-west-palm-beach-va-medical-center",
  "fieldFacilityLocatorApiId": "vba_317a",
  "id": "a08da437-55a4-451c-8f90-af498f38a297",
  "image": null,
  "lastUpdated": "2023-09-19T13:13:04+00:00",
  "metatags": [
    {
      "attributes": {
        "content": "Veteran Readiness and Employment Office at West Palm Beach VA Medical Center | Veterans Affairs",
        "name": "title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "We help Veterans, service members, and their families access VA benefits and services. Benefits we can help with include disability compensation, education benefits, life insurance, pensions, and home loans. Find a benefit office near you or sign up for updates.",
        "name": "description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "href": "https://www.va.gov/img/design/logo/va-og-image.png",
        "rel": "image_src",
      },
      "tag": "link",
    },
    {
      "attributes": {
        "content": "Veterans Affairs",
        "property": "og:site_name",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Veteran Readiness and Employment Office at West Palm Beach VA Medical Center | Veterans Affairs",
        "property": "og:title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "We help Veterans, service members, and their families access VA benefits and services. Benefits we can help with include disability compensation, education benefits, life insurance, pensions, and home loans. Find a benefit office near you or sign up for updates.",
        "property": "og:description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "https://www.va.gov/img/design/logo/va-og-image.png",
        "property": "og:image",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "U.S. Department of Veterans Affairs",
        "property": "og:image:alt",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "summary_large_image",
        "name": "twitter:card",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "@DeptVetAffairs",
        "name": "twitter:site",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Veteran Readiness and Employment Office at West Palm Beach VA Medical Center | Veterans Affairs",
        "name": "twitter:title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "https://www.va.gov/img/design/logo/va-og-image.png",
        "name": "twitter:image",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "U.S. Department of Veterans Affairs",
        "name": "twitter:image:alt",
      },
      "tag": "meta",
    },
  ],
  "moderationState": "archived",
  "officeHours": [
    {
      "all_day": false,
      "comment": "",
      "day": 1,
      "endhours": 1630,
      "starthours": 800,
    },
    {
      "all_day": false,
      "comment": "",
      "day": 2,
      "endhours": 1630,
      "starthours": 800,
    },
    {
      "all_day": false,
      "comment": "",
      "day": 3,
      "endhours": 1630,
      "starthours": 800,
    },
    {
      "all_day": false,
      "comment": "",
      "day": 4,
      "endhours": 1630,
      "starthours": 800,
    },
    {
      "all_day": false,
      "comment": "",
      "day": 5,
      "endhours": 1630,
      "starthours": 800,
    },
    {
      "all_day": false,
      "comment": "Closed",
      "day": 6,
      "endhours": null,
      "starthours": null,
    },
    {
      "all_day": false,
      "comment": "Closed",
      "day": 0,
      "endhours": null,
      "starthours": null,
    },
  ],
  "operatingStatusFacility": "normal",
  "operatingStatusMoreInfo": null,
  "phoneNumber": "800-827-1000",
  "prepareForVisit": [],
  "published": false,
  "timezone": "America/New_York",
  "title": "Veteran Readiness and Employment Office at West Palm Beach VA Medical Center",
  "type": "node--vba_facility",
}
`;
